src/galois/_polys/_primitive.py

```diff
@@ -86,35 +115,59 @@ def primitive_poly(order: int, degree: int, method: Literal["min", "max", "rando
     """
     verify_isinstance(order, int)
     verify_isinstance(degree, int)
+    verify_isinstance(terms, (int, str), optional=True)
 
     if not is_prime_power(order):
         raise ValueError(f"Argument 'order' must be a prime power, not {order}.")
     if not degree >= 1:
         raise ValueError(
             f"Argument 'degree' must be at least 1, not {degree}. There are no primitive polynomials with degree 0."
         )
+    if isinstance(terms, int) and not 1 <= terms <= degree + 1:
+        raise ValueError(f"Argument 'terms' must be at least 1 and at most {degree + 1}, not {terms}.")
+    if isinstance(terms, str) and not terms in ["min"]:
+        raise ValueError(f"Argument 'terms' must be 'min', not {terms!r}.")
     if not method in ["min", "max", "random"]:
         raise ValueError(f"Argument 'method' must be in ['min', 'max', 'random'], not {method!r}.")
 
-    if method == "min":
-        poly = next(primitive_polys(order, degree))
-    elif method == "max":
-        poly = next(primitive_polys(order, degree, reverse=True))
-    else:
-        poly = _random_search(order, degree)
+    try:
+        if method == "min":
+            poly = next(primitive_polys(order, degree, terms))
+        elif method == "max":
+            poly = next(primitive_polys(order, degree, terms, reverse=True))
+        else:
+            if terms == "min":
+                terms = _minimum_terms(order, degree, "is_primitive")
+            poly = _random_search(order, degree, terms)
```

Comment:
What do you think about moving the `if terms == "min"` to the top? I would like to only include the code to search in the database in one place, but as it is, if a user writes, for example, `primitive_poly(7, 129, terms="min")`, it would still enter the first `if method="min"`.

Reply:
I have a commit I'm about to push. Can you let me know your suggested change to my latest?

Reply:
As an alternative, you could also change the default value of `method` to `"random"` when the user specifies `terms="min"`. What do you think?

Reply:
This was the kind of thing I had in mind for the database lookup. What are your thoughts?

```python
    try:
        if terms == "min" and method == "min":
            try:
                # Database lookup
                return poly
            except LookupError:
                pass

        if method == "min":
            return next(irreducible_polys(order, degree, terms))
        if method == "max":
            return next(irreducible_polys(order, degree, terms, reverse=True))

        # Random search
        if terms is None:
            return next(_random_search(order, degree, "is_irreducible"))
        if terms == "min":
            terms = _minimum_terms(order, degree, "is_irreducible")
        return next(_random_search_fixed_terms(order, degree, terms, "is_irreducible"))
```

Reply:
That works too, yes!

Reply:
Actually, even better is this:

```python
    if terms is "min" and method == "min":
        try:
            # Database lookup
            return poly
        except LookupError:
            pass

    try:
        if method == "min":
            return next(irreducible_polys(order, degree, terms))
        if method == "max":
            return next(irreducible_polys(order, degree, terms, reverse=True))

        # Random search
        if terms is None:
            return next(_random_search(order, degree, "is_irreducible"))
        if terms == "min":
            terms = _minimum_terms(order, degree, "is_irreducible")
        return next(_random_search_fixed_terms(order, degree, terms, "is_irreducible"))

    except StopIteration as e:
        terms_str = "any" if terms is None else str(terms)
        raise RuntimeError(
            f"No monic irreducible polynomial of degree {degree} over GF({order}) with {terms_str} terms exists."
        ) from e
```